Source/MQTTnet/Formatter/MqttBufferReader.cs

```diff
@@ -45,32 +44,36 @@ public byte ReadByte()
         {
             ValidateReceiveBuffer(1);
 
-            return _buffer[_position++];
+            return _buffer[Position++];
         }
 
         public uint ReadFourByteInteger()
         {
             ValidateReceiveBuffer(4);
 
-            var byte0 = _buffer[_position++];
-            var byte1 = _buffer[_position++];
-            var byte2 = _buffer[_position++];
-            var byte3 = _buffer[_position++];
+            var byte0 = _buffer[_offset + Position];
```

Comment:
FYI: for "modern targets" this could be done via `BinaryPrimitives.ReadUInt32BigEndian`-method.

Reply:
I adopted this but enforced the reverse order. With the method you mentioned it depends on the CPU.

Reply:
Ah, right that "cleans up" the order automatically. `MemoryMarshal.ReadXYZ` would be the correct one. 
The gain is one memory read instead of the individual bytes (which wil most likely be in the same cacheline). Effect will be only 1 bound-check instead of the 4.